docs/HighLevelDesign.md

```diff
@@ -81,3 +81,38 @@ their subgraph.
 * [Add a new graph
 transform](../include/onnxruntime/core/optimizer/graph_transformer.h)
 * [Add a new rewrite rule](../include/onnxruntime/core/optimizer/rewrite_rule.h)
+
+## The ONNX Runtime and Windows OS integration
+
+The ONNX runtime shipped with the Windows operating system in build 1809 (RS5).  The runtime was embedded inside the Windows.AI.MachineLearning.dll and was exposed via that WinRT API (WinML for short).  It includes CPU support and a DirectML execution provider for GPU support.   Since then it has continued to ship in every version of Windows.
+
+Starting with the 1.2 release we are bringing a new layered architecture to the ONNX Runtime and Windows ML.
+*Note:  This feature is preview as of the 1.2 release*
+
+The high level design looks like this
+
+![ONNX + WinML layered architecture](images/layered-architecture.png)
+
+You can see we replaced the embedded ONNX runtime with the new ONNXRuntime.dll.  With this new approach customers have flexibility on which API they choose to use and on how they want to distribute the binaries.
+
+### API choice
+
+Developers can now choose which API works best for their scenario.
+
+||WinRT|C API|
+|--|--|--|
+|Type system| Integration with Windows RT types| Platform neutral types|
+|Lanugage support| Language support via WinRT Projections| Language support via per language projections|
+|Tensorization| Accepts VideoFrame's and converts to tensors (support for CPU and GPU)| Accepts tensors|
```

Comment:
plural

Reply:
:)

---
In reply to: [390704139](https://github.com/microsoft/onnxruntime/pull/3149#discussion_r390704139) [](ancestors = 390704139)